pkg/plugin/run.go

```diff
@@ -150,6 +149,13 @@ func (e *AutoscaleEnforcer) handleAgentRequest(
 		return nil, 400, fmt.Errorf("nil metrics not supported for protocol version %v", req.ProtoVersion)
 	}
 
+	// Check that req.ComputeUnit == nil matches whether the protocol version supports it.
+	if req.ComputeUnit == nil && req.ProtoVersion.AgentSendsComputeUnit() {
+		return nil, 400, fmt.Errorf("computeUnit required for protocol version %v", req.ProtoVersion)
```

Comment:
We can use `http.StatusBadRequest` instead of 400 but better to be consistent in the function. So leave it for now.

Reply:
Long ago, I had this conversation with @﻿tychoish, and we ended up keeping the "common" numbers because they're easily recognizable (and IMO, end up being more readable because of it).

If you'd like, we can revisit this. I'm slight-negative on making that change, but ok to go with team consensus.

Reply:
I see. I'm a fan of using constants myself. We can leave this for now and discuss it later.